File: demo/HelloSwift/HelloSwift/Sort/InsertionSort.swift
//
//  InsertionSort.swift
//  HelloSwift
//
//  Created by muhlenXi on 2019/9/26.
//  Copyright © 2019 muhlenXi. All rights reserved.
//

import Foundation

func insertionSort(array: inout[Int]) {
    guard array.count > 1 else {
        return
    }
    
    for index in 1..<array.count {
        var currentIndex = index
        while currentIndex > 0 {
            if array[currentIndex-1] > array[currentIndex] {
                let temp = array[currentIndex-1]
                array[currentIndex-1] = array[currentIndex]
                array[currentIndex] = temp
            }
            currentIndex -= 1
        }
    }
}

File: demo/HelloJava/src/Sort/InsertionSort.java
package Sort;

public class InsertionSort {
    public static void main() {
        int[] array = {4, 2, 5, 3, 1};

        for (int i:array) {
            System.out.println(i);
        }

        insertionSort(array);

        for (int i:array) {
            System.out.println(i);
        }
    }

    static void insertionSort(int[] array) {
        for(int i = 0; i < array.length;i ++) {
            int currentIndex = i;
            while(currentIndex > 0) {
                if(array[currentIndex-1] > array[currentIndex]){
                    int temp = array[currentIndex-1];
                    array[currentIndex-1] = array[currentIndex];
                    array[currentIndex] = temp;
                }
                currentIndex -= 1;
            }
        }
    }
}
